src/errors.rs

```diff
@@ -0,0 +1,79 @@
+use std::{self, error, fmt, io, str};
+use semver::{self, Identifier};
+
+/// The error type for this crate.
+#[derive(Debug)]
+pub enum Error {
+    /// An error ocurrend when executing the `rustc` command.
+    CouldNotExecuteCommand(io::Error),
+    /// The output of `rustc -vV` was not valid utf-8.
+    Utf8Error(str::Utf8Error),
+    /// The output of `rustc -vV` was not in the expected format.
+    UnexpectedVersionFormat,
+    /// An error ocurred in parsing a `VersionReq`.
+    ReqParseError(semver::ReqParseError),
+    /// An error ocurred in parsing the semver.
+    SemVerError(semver::SemVerError),
+    /// The pre-release tag is unknown.
+    UnknownPreReleaseTag(Identifier),
+}
```

Comment:
I'm not opposed to add this as is, but I feel like half of these error scenarios could reasonably be declared unexpected failures that don't need to appear in the Result. :)

Eg, rustc changing to non-utf8 output or not using semver compatible version numbers is unlikely to be an error, ever.

Though I guess since a) there are already many error variants in that enum and b) `RUSTC` might be a wrapper or alternate compiler, just covering everything is probably for the best anyway.